src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java

```diff
@@ -136,6 +137,16 @@ public void onPortalEnter(EntityPortalEnterEvent e) {
         }
     }
 
+    // Fixing Issue #2351
+    @EventHandler
+    public void onLeash(PlayerLeashEntityEvent e) {
+        if (!isEnabled()) {
+            return;
+        }
+
+        e.setCancelled(true);
+    }
```

Comment:
Ehm, that just disables leashing as a whole...
So noone would be able to leash anything anymore.
You would still need to check if someone used a grappling hook.

Reply:
Oh okok sorry I thought that you had isolated the event! Let me try it then ;)

Reply:
Nope, event isolation sadly isn't a thing in the Bukkit API XD
You'll still need to add manual checks for things

Reply:
@TheBusyBiscuit just a question, the CI is saying that could not found PlayerLeashEntityEvent should I use another event?

Reply:
You imported it from the wrong namespaced, that's why.
It's located in the `org.bukkit.event.entity` package.
Source: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/entity/PlayerLeashEntityEvent.html